docs/advanced/pycpp/object.rst

```diff
@@ -33,19 +33,91 @@ The reverse direction uses the following syntax:
 
 When conversion fails, both directions throw the exception :class:`cast_error`.
 
+.. _python_libs:
+
+Accessing Python libraries from C++
+===================================
+
+It is also possible to import types and classes defined in the Python standard
+library or available in the current Python environment (``sys.path``) and work
+with these in C++.
+
+This example obtains a reference to the Python ``Decimal`` class.
+
+.. code-block:: cpp
+
+    // Equivalent to "from decimal import Decimal"
+    py::object Decimal = py::module::import("decimal").attr("Decimal");
+
+.. code-block:: cpp
+
+    // Try to import scipy
+    py::object scipy = py::module::import("scipy");
+    return scipy.attr("__version__");
+
 .. _calling_python_functions:
 
 Calling Python functions
 ========================
 
-It is also possible to call python functions via ``operator()``.
+It is also possible to call Python, classes, functions and methods 
+via ``operator()``.
+
+.. code-block:: cpp
+
+    // Construct a Python object of class Decimal
+    py::object pi = Decimal("3.14159");
+
+.. code-block:: cpp
+
+    // Use Python to make our directories
+    py::object os = py::module::import("os");
+    py::object makedirs = os.attr("makedirs");
+    makedirs("/tmp/path/to/somewhere");
+
+One can convert the result obtained from Python to a pure C++ version 
+if a ``py::class_`` or type conversion is defined.
 
 .. code-block:: cpp
 
     py::function f = <...>;
     py::object result_py = f(1234, "hello", some_instance);
     MyClass &result = result_py.cast<MyClass>();
 
+.. _calling_python_methods:
+
+Calling Python methods
+========================
+
+To call a object's method, one can again use ``.attr`` to obtain access to the
+Python method.
+
+.. code-block:: cpp
+
+    // Calculate e^π in decimal
+    py::object pi_squared = pi.attr("exp")();
+    std::cout << py::str(pi_squared);
+
+In the example above ``pi.attr("exp")`` is a *bound method*: it will always call
+for the method for that same instance of the class. (The first argument of the
+method, by convention ``self``, is bound to ``pi``, even if called for other
```

Comment:
To me, the wording here with "even if" sounds like it could be called with a different `self` but it would be ignored. Consider rephrasing to something like "it's implicit and doesn't appear alongside other call arguments". (Although, readers should already be familiar with this from pure Python.)

Reply:
Removed sentence. I agree anyone using pybind11 is probably familiar with this but it's not that commonly used in Python; mentioning bound/unbound methods should be enough of a clue.